src/utils/http/index.ts

```diff
@@ -0,0 +1,33 @@
+import axios from 'axios'
+
+import type { AxiosResponse } from 'axios'
+
+const BASE_URL = 'https://offer-be.kro.kr'
+
+const Axios = axios.create({
+  baseURL: BASE_URL,
+  headers: {
+    'content-type': 'application/json'
+  }
+})
+
+const getResult = <T>(response: AxiosResponse<T>): T => response.data
+
+export const http = {
+  get: <Response = unknown>(url: string): Promise<Response> =>
+    Axios.get<Response>(url).then(getResult),
+  post: <Response = unknown, Request = unknown>(
+    url: string,
+    data?: Request
+  ): Promise<Response> => Axios.post<Response>(url, data).then(getResult),
+  put: <Response = unknown, Request = unknown>(
+    url: string,
+    data?: Request
+  ): Promise<Response> => Axios.put<Response>(url, data).then(getResult),
+  patch: <Response = unknown, Request = unknown>(url: string, data?: Request) =>
+    Axios.patch<Response>(url, data).then(getResult),
+  delete: <Response = unknown, Request = unknown>(
+    url: string,
+    data?: Request
+  ): Promise<Response> => Axios.delete<Response>(url, { data }).then(getResult)
```

Comment:
```suggestion
  post: <Response = unknown, Request = unknown>(
    url: string,
    payload?: Request
  ): Promise<Response> => Axios.post<Response>(url, payload).then(getResult),
  put: <Response = unknown, Request = unknown>(
    url: string,
    payload?: Request
  ): Promise<Response> => Axios.put<Response>(url, payload).then(getResult),
  patch: <Response = unknown, Request = unknown>(url: string, payload?: Request) =>
    Axios.patch<Response>(url, payload).then(getResult),
  delete: <Response = unknown, Request = unknown>(
    url: string,
    payload?: Request
  ): Promise<Response> => Axios.delete<Response>(url, { data: payload }).then(getResult)
```

쓰기 동작할 때의 파라미터를 보통 payload로 표현해서 명시적으로 구분될 수 있게 수정이 되면 좋을 것 같은데 어떠신가요?
- 파라미터: payload
- 리턴값: data

+ 왜 delete메서드만 data로 감싸서 전달하는 걸까요?!

Reply:
* payload 좋슴다 !
* axios 타입을 보면 delete는 data를 config로 전달해줘야 함두 ! 
<img width="1098" alt="스크린샷 2023-11-14 오후 10 48 58" src="https://github.com/price-offer/offer-fe/assets/70738281/b4324bdf-5d01-450b-919c-0383deecc941">

Reply:
[반영 완료](https://github.com/price-offer/offer-fe/pull/164/commits/bd0a5beffa7b9d2ff1ee6d0afa3b8ee60b6d37be) !

Reply:
@shinhyojeong 오잉? 다 같은 타입 아닌가유???

Reply:
@sonsurim 앗 매개변수를 보시면 됩니다 ! delete매서드의 매개변수에는 data가 없어서 config를 통해서 전달해야해용

Reply:
아! 확인했습니당 감사합니둥 ㅎㅎ
delete는 보통 get처럼 url로 처리하는데 인터페이스를 put, patch, post와 같은 형태로 가져가야 하는지도 의견 여쭤봅니당

Reply:
홍뇽 같게 처리하는 것이 좋을 것 같네요 get처럼 변경하겠슴다!

Reply:
[반영 완료](https://github.com/price-offer/offer-fe/pull/164/commits/ed8981d7004634c7e162670b91242fa6bc0c2930) !